### DISCUSSION TITLE
Understanding the DebugPlugin's Functionality

### DISCUSSION THREAD

**Discussion by kkopp001** [swup/swup#808] [Help] [answered]
Hi !

I recently started using the DebugPlugin with swup, and I'm trying to understand its full capabilities. In the past, I've used Barba, and its debug feature was quite helpful. For instance, if there was a JavaScript error, like an uninitialized variable when using GSAP, Barba's debug would point it out in the console.

However, with SWUP's DebugPlugin, I've noticed it provides information about the lifecycle events, like what I clicked on, scroll positions, page loads, etc. But when I encountered a JavaScript error (e.g., forgetting to import GSAP), the DebugPlugin didn't notify me of this error in the console.

Could someone clarify how the DebugPlugin is intended to work? Specifically, does it handle JavaScript errors, or is its primary function to provide insights into SWUP's lifecycle events? Any guidance on how to effectively use this plugin for debugging would be greatly appreciated.

Thank you!

**Comment by hirasso** [comment, marked as answer]
Hi @kkopp001 

JavaScript errors in your code should always be thrown in your console. This is just native browser behavior.

@swup/debug-plugin is intended to help with understanding swup's page visit lifecycle events, as you described above. It won't interfere with the runtime in any way. It's a purely informational tool.

**Comment by kkopp001** [comment]
Alirght I see thank you !

**Reply by hirasso** [reply]
No worries! Feel free to mark the answer as answer :)
